ebml_matroska.xml

```diff
@@ -332,7 +332,21 @@
   <element name="BlockAddIDType" path="\Segment\Tracks\TrackEntry\BlockAdditionMapping\BlockAddIDType" id="0x41E7" type="uinteger" minOccurs="1" maxOccurs="1" minver="4" default="0">
     <documentation lang="en" purpose="definition">Stores the registered identifer of the Block Additional Mapping to define how the BlockAdditional data should be handled.</documentation>
     <extension webm="0"/>
-  </element>
+    <restriction>
+      <enum value="0" label="BlockAddIDValue defines the content type">
+        <documentation lang="en" purpose="definition">This value indicates that BlockAddIDValue defines the content type, and a list of content type is in BlockAddID documentation.</documentation>
+      </enum>
+      <enum value="1" label="Codec specific content">
+        <documentation lang="en" purpose="definition">This value indicates that BlockAddIDExtraData defines the content type, and this definition depends on the Codec ID used. A transcoder SHOULD discard this content if the addition is not supported</documentation>
```

Comment:
Why place any restriction/recommendation on whether this data should be kept or discarded? Why should a demuxer/remuxer know whether such data is important? I'd leave that out, or always state that "data SHOULD be kept" (obviously data might be discarded on user request, but a generic remuxer should keep as much data as possible). Maybe the extension was registered in a later version of the specs and the remuxer doesn't know about it. Or maybe it's a generic demuxer that only passes data along to the codec, and it's up to the codec to decide if the data is actually relevant.

A specialized tool for enforcing format compatibility might make other choices, true, but this text is about the general case.